src/lib.rs

```diff
@@ -137,19 +151,389 @@ pub fn get_verifiable_credential_public_key(
 ) -> Result<String, ConcordiumWalletCryptoError> {
     get_verifiable_credential_public_key_aux(seed_hex, net.as_str(), issuer_index, issuer_subindex, verifiable_credential_index)
         .map_err(|e| ConcordiumWalletCryptoError::CallFailed {
-            call: format!("get_verifiable_credential_public_key(seed_hex, net={net}, issuer_index={issuer_index}, issuer_subindex={issuer_subindex}, verifiable_credential_index={verifiable_credential_index})"),
-            msg: e.to_string(),
+            call: format!("verifiable_credential_public_key_hex(seed_hex, net={net}, issuer_index={issuer_index}, issuer_subindex={issuer_subindex}, verifiable_credential_index={verifiable_credential_index})"),
+            msg: format!("{:#}", e),
         })
 }
 
-pub fn get_verifiable_credential_backup_encryption_key(
+pub fn verifiable_credential_backup_encryption_key_hex(
     seed_hex: String,
     net: String,
 ) -> Result<String, ConcordiumWalletCryptoError> {
     get_verifiable_credential_backup_encryption_key_aux(seed_hex, net.as_str()).map_err(|e| {
         ConcordiumWalletCryptoError::CallFailed {
-            call: format!("get_verifiable_credential_backup_encryption_key(seed_hex, net={net}"),
-            msg: e.to_string(),
+            call: format!("verifiable_credential_backup_encryption_key_hex(seed_hex, net={net}"),
+            msg: format!("{:#}", e),
         }
     })
 }
+
+pub type ArIdentity = u32;
+pub type IpIdentity = u32;
+pub type KeyIndex = u8;
```

Comment:
Why do these definitions need to be duplicated?

Reply:
Turns out that because the types in concordium-base wraps the types (in contrast to being type aliases) complicates their use in a way that isn't worth it. Besides, these aliases is a rather arbitrary subset of the used types anyway, so I inlined them instead.

Reply:
done